File: src/main/java/com/fifo/ticketing/domain/book/dto/BookedView.java
package com.fifo.ticketing.domain.book.dto;

import com.fifo.ticketing.domain.book.entity.BookStatus;
import java.util.List;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class BookedView {

    private Long bookId;
    private Long performanceId;
    private String performanceTitle;
    private String encodedFileName;
    private String placeName;
    private List<BookSeatViewDto> seats;
    private int totalPrice;
    private int quantity;
    private BookStatus bookStatus;
}

File: src/main/java/com/fifo/ticketing/domain/book/dto/BookedView.kt
package com.fifo.ticketing.domain.book.dto

import com.fifo.ticketing.domain.book.entity.BookStatus

data class BookedView(
    val bookId: Long,
    val performanceId: Long,
    val performanceTitle: String,
    val encodedFileName: String,
    val placeName: String,
    val seats: List<BookSeatViewDto>,
    val totalPrice: Int = 0,
    val quantity: Int = 0,
    val bookStatus: BookStatus,
    val urlPrefix: String,

    ) {
    fun getUrl(): String = "$urlPrefix$encodedFileName"
}
